pkg/userdata/ubuntu/userdata.go

```diff
@@ -117,18 +119,18 @@ func (p Provider) UserData(
 		return "", fmt.Errorf("error extracting server address from kubeconfig: %v", err)
 	}
 
-	var crPkg, crPkgVersion string
-	if spec.Versions.ContainerRuntime.Name == containerruntime.Docker {
-		crPkg, crPkgVersion, err = getDockerInstallCandidate(spec.Versions.ContainerRuntime.Version)
-		if err != nil {
-			return "", fmt.Errorf("failed to get docker install candidate for %s: %v", spec.Versions.ContainerRuntime.Version, err)
-		}
-	} else {
-		return "", fmt.Errorf("unknown container runtime selected '%s'", spec.Versions.ContainerRuntime.Name)
+	if pconfig.ContainerRuntimeInfo.Name != containerruntime.Docker {
```

Comment:
Just as a idea:
Should we remove the possibility to specify the docker-version via Machine manifest?
IMO the controller should be very opinionated here. It would make our code a bit easier.
But this has nothing to do with this PR

Reply:
Sounds like a good idea but I'd prefer not to do that within this pull, its already pretty huge

Reply:
Yes, definitely something for a follow up!